backend/src/services/model.ts

```diff
@@ -420,9 +420,18 @@ export async function saveImportedModelCard(modelCard: ModelCardRevisionInterfac
 }
 
 export async function setLatestImportedModelCard(modelId: string) {
-  // Add check for mirrored model
-  const latestModelCard = await ModelCardRevisionModel.findOne({ modelId }).sort({ version: -1 }).limit(1)
-  await ModelModel.updateOne({ id: modelId }, { $set: { card: latestModelCard } })
+  const latestModelCard = await ModelCardRevisionModel.findOne({ modelId }).sort({ version: -1 })
+  if (!latestModelCard) {
+    throw NotFound('Cannot find latest model card.')
+  }
+
+  const result = await ModelModel.findOneAndUpdate(
+    { id: modelId, 'settings.mirror.sourceModelId': { $exists: true, $ne: '' } },
+    { $set: { card: latestModelCard } },
+  )
+  if (!result) {
+    throw InternalError('Unable update model with latest model card ID.')
```

Comment:
Could just be me, but this message is a little confusing. The message refers to "latest model card ID", but we're not using that ID in the query above.